tests/test_meshtal_analysis.py

```diff
@@ -70,8 +70,80 @@ def test_find_mesh_tally_num(self):
         data = ut.get_lines(path)
         
         self.assertEqual(meshtal_analysis.find_mesh_tally_numbers(data), {214: 4})
-            
+ 
+          
+class add_mesh_test(unittest.TestCase):
+     
+    def test_add_mesh_error(self):
+        mesh1_test = meshtal_analysis.meshtally()
+        mesh1_test.ctype="6col"
+        mesh2_test = meshtal_analysis.meshtally()
+        mesh2_test.ctype="6col"
+        
+        mesh1_test.x_bounds = (-9.0, -9.1)
+        mesh1_test.y_bounds = (-9.0, -9.1)
+        mesh1_test.z_bounds = (1.4, 1.5)
+        mesh2_test.x_bounds = (9.0, 9.1)
+        mesh2_test.y_bounds = (9.0, 9.1)
+        mesh2_test.z_bounds = (7.8, 7.9)
+        
+        with self.assertRaises(ValueError):
+            meshtal_analysis.add_mesh(mesh1_test, mesh2_test)
+        
+    def test_add_mesh(self):
+        mesh3_test = meshtal_analysis.meshtally()
+        mesh3_test.ctype="6col"
+        mesh4_test = meshtal_analysis.meshtally()
+        mesh4_test.ctype="6col"
+        
+        mesh3_test.data = [['1.000E+36', '-9.0', '-9.0', '1.4', '7.329430e-07', '0.017765']]
+        mesh4_test.data = [['1.000E+36', '-9.0', '-9.0', '1.4', '6.566330e-07', '0.018680']]
+        mesh3_test.data = meshtal_analysis.convert_to_df(mesh3_test)
+        mesh4_test.data = meshtal_analysis.convert_to_df(mesh4_test)
+        
+        mesh3_test.x_bounds = (-9, -9.1)
```

Comment:
the xyz values in the .data variable should be mid points, but you have the same values in your bounds